luxonis_ml/data/datasets/luxonis_dataset.py

```diff
@@ -292,6 +293,9 @@ def set_classes(self, classes: List[str], task: Optional[str] = None) -> None:
             self.fs.put_file(local_file, "metadata/classes.json")
             self._remove_temp_dir()
 
+    def set_global_metadata(self, metadata: Dict[str, Any]) -> None:
+        self.global_metadata = metadata
```

Comment:
Due to GCS datasets, I think we need a way to persist this via storage instead of just memory? Perhaps we could use the existing `datasets.json` or `metadata` folder?